ui/src/app/routing/guards.tsx

```diff
@@ -40,15 +44,41 @@ export const userDisabledPageGuard = (userDisabledInDb: boolean): Guard => ({
   redirectPath: '/',
 });
 
-export const registrationGuard: Guard = {
-  allowed: (): boolean => hasRegisteredTierAccess(profileStore.get().profile),
-  redirectPath: DATA_ACCESS_REQUIREMENTS_PATH,
+const allCompleteOrBypassed = (
+  profile: Profile,
+  moduleNames: AccessModule[]
+) => {
+  const modules = profile?.accessModules?.modules;
+  return moduleNames.every((moduleName) => {
+    const status = getAccessModuleStatusByNameOrEmpty(modules, moduleName);
+    return !!status?.completionEpochMillis || !!status?.bypassEpochMillis;
+  });
+};
+
+// use this for all access-module routing decisions, to ensure only one routing is chosen
+export const shouldRedirectToMaybe = (profile: Profile): string | undefined => {
+  return cond(
+    [profile?.accessModules?.anyModuleHasExpired, () => ACCESS_RENEWAL_PATH],
+    // not a common scenario (mainly test users) but AAR is the only way to recover if these modules are missing
+    [
```

Comment:
This section is the only real change: PROFILECONFIRMATION-incomplete and PUBLICATIONCONFIRMATION-incomplete users will now be routed to AAR where they can complete these modules.